### DISCUSSION TITLE
Bend direction

### DISCUSSION THREAD

**Discussion by HYBRID-BEING** [shaise/FreeCAD_SheetMetal#277] [Q&A]
Hello. i'm probably using Sheet Metal WB in a way not really intended, but bear with me.
I've got a little PCB (composite AV adapter for Virtual IO i-glasses) with a rather inconvenient array of through-holes:
```
o o o o o
  o o
(pitch is 3.5mm horizontally and 5mm vertically)
```
A cable was soldered directly to them, and it was in a rather sorry state, so i had to desolder it and patch it up[^1]. In case i would need to do that again, i wanted to use some kind of connector, but i'm tight on space, and i'm kinda limited in purchasing options, so what i thought i'd do, was to get long header pins, bend them around and package them in a tight shell (3d-printed).
On the other side of a cable there is a 2x2 pitch 8pin female connector (with 7 pins wired), so the pinout should, obviously, match. This, coupled with a need to adapt pitch, complicates all the bends even further, and i was hoping i could use SM to prototype a package around this mess.

At first, it looked like SM had not trouble dealing with a square-profile (0.5mm side) pins, and everything was going fine and dandy, until i had to make a bend very close (0.01mm) to another one[^2]. I was making bends by selecting an edge (which to me looks counter-intuitive, cause it makes a bend in a direction opposite of said edge, though i digress), but in this case SM chose to base a bend not on a pin profile face, but on an adjacent face (smaller one).
I then tried to make a bend by selecting a profile face, but SM chose a direction i didn't want - it made a bend in the same plane as the first one, but i needed to bend a pin in a perpendicular plane.
I wasn't able to figure out if there is a way to control both of these behaviors. I guess it's not really an issue when you are working with metal sheets, hmm...

I did found a way to circumvent that by using Fold a Wall (SMFoldWall) method instead, but it seems to have bugs with length calculation and, surprisingly, is a discouraged design method (#122).

[^1]: The cable is kinda unusual - in a single tube it has 2 wires and a braid in one insulation, a wire and a braid in another insulation, and another two wires. In theory, i could probably procure some individually braided twisted pair cable, but since it's video and audio signal, i decided it's best to keep it original.
[^2]: What i really wanted was to twist a pin before the bend, so after a 90deg bend it would go sideways at a slight angle, but, alas, there is no twist tool in SM ([an idea for a feature?](https://www.youtube.com/watch?v=1rSHS6gknPk)). So, i resorted to making two consecutive Bends (SMMakeWall) with 0.01 Extend (SMExtrudeFace) in-between.

**Comment by shaise** [comment]
indeed working with square faces can be problematic as when you fold on a face, SM guesses the direction of the bend by assuming the longer edge to be on the face of the sheet, and the shorter is on the sheet sides. One way to overcome it is by slightly cheating. Instead of using exact square face, use a 0.5 x 0.5001 face, this way forcing SM to bend along the 0.5 side.
